FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs

```diff
@@ -66,7 +67,72 @@ public class AzureBlobJobStore : IJobStore
             _renewLockTimer.Elapsed += async (sender, e) => await RenewJobLockLeaseAsync();
         }
 
-        public async Task<bool> AcquireJobLock(CancellationToken cancellationToken = default)
+        public async Task<Job> AcquireJobAsync(CancellationToken cancellationToken = default)
+        {
+            var lockAcquired = await TryAcquireJobLockAsync(cancellationToken);
+            if (!lockAcquired)
+            {
+                _logger.LogWarning("Start job conflicted. Failed to acquire job lock.");
```

Comment:
Maybe ```LogError``` here because next line will throw an exception?

Reply:
Yeah, for now we should log an error as exception is raised. Thanks